docs/src/docs/asciidoc/end-to-end-gradle-guide.adoc

```diff
@@ -0,0 +1,871 @@
+
+So if you modified existing metadata file(s) on the default location, please do the following:
+
+1. Move your metadata to some other, non-default location (with the `metadataCopy` task for example)
+2. Set `mergeWithExisting` option to true in the `metadataCopy` block
+3. Run your tests again to generate new metadata (as we already described in the <<collect-metadata, collect metadata section>>)
+4. Run `metadataCopy` task again (with `mergeWithExisting` set to true in step 2)
+
+This way you will keep your original metadata, and add a new one.
+
+
+[[reachability-metadata-repository]]
+== Reachability metadata repository
+
+Native Build Tools (both Gradle and Maven plugins) picks metadata from Reachability metadata repository to ensure your application works out-of-box (if all metadata required by your app is already contributed to the metadata repository).
+Furthermore, you can configure Reachability metadata support through `metadataRepository` block added to our main `graalvmNative` block inside `build.gradle`.
+Most common options you may want to configure in this block are:
+
+* `enabled` - determines if you want to use Reachability metadata support or not (`true` by default)
+* `version` - specifies exact Reachability metadata version you want to use
+* `uri` - specifies the url where the metadata is stored. This can be used to point to the local repository
+
+YYou can read more about __Reachability metadata support__ and other (advanced) configuring options, https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#_configuring_the_metadata_repository[here].
+
+In some cases, when you want to maintain multiple projects that share common metadata from various libraries, you should consider contributing metadata to https://github.com/oracle/graalvm-reachability-metadata[Reachability metadata project].
+https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md[Contributing to the repository] should be simple:
+
+- Clone repository locally:
+
+[source,bash,subs="verbatim,attributes", role="multi-language-sample"]
+----
+git clone git@github.com:oracle/graalvm-reachability-metadata.git
+----
+
+- generate metadata and test stubs (replace with the GAV coordinates of library you are providing metadata for):
+
+[source,bash,subs="verbatim,attributes", role="multi-language-sample"]
+----
+./gradlew scaffold --coordinates com.example:my-library:1.0.0
+----
+
+- implement tests in test stubs that will show how you have generated metadata
+- collect metadata as described https://github.com/oracle/graalvm-reachability-metadata/blob/master/docs/CollectingMetadata.md#collecting-metadata-for-a-library[here]
+- create a pull request and fill the checklist
+
+[[track-diagnostics]]
+== Track diagnostics
+
+If you want to explore details about native images you are generating, you can add:
+
+[source,bash,subs="verbatim,attributes", role="multi-language-sample"]
+----
+buildArgs.add("--emit build-report")
```

Comment:
:?: if it is really `--emit build-report`, then it should be 2 build args: `--emit`, `build-report`.

Reply:
Of course we can also separate this and it will work, but in my opinion it is more understandable if we group argument with its value. It will work both ways.